mlflow/utils/requirements_utils.py

```diff
@@ -180,16 +180,31 @@ def _run_command(cmd):
         raise MlflowException(msg)
 
 
-def _get_package_version_from_metadata(package):
+def _get_installed_version(package, module=None):
     """
-    Obtains the version of the specified package from its metadata.
+    Obtains the package version using `importlib_metadata.version`. If it fails, use
+    `__import__(module or package).__version__`.
     """
-    version = importlib_metadata.version(package)
+    try:
+        version = importlib_metadata.version(package)
```

Comment:
If a user installs `torch` with:

```
pip install torch==1.8.0+cpu -f https://download.pytorch.org/whl/torch_stable.html
```

`importlib_metadata.version("torch")` returns `1.8.0`.

Reply:
Or we could just log `torch==1.8.0+cpu`. If the user wants to specify `--find-links`,  he/she can use `pip_requirements`.

Reply:
@harupy Ideally, we'd like to maximize the likelihood of serving succeeding. Can we ignore local version identifiers and print a warning indicating that the local version identifier was dropped & instructing customers to use `pip_requirements` or `conda_env` to specify the distribution containing the local version if necessary?

Reply:
Ok, I'll update the code to drop local version identifiers and raise a warning for that.